### PR TITLE
LVNOD-44_Create_Profile_model,_migration_and_seeders

### PR DESCRIPTION
**Pull request by yuliannakorol**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-26)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-44)

## Code reviewers

- [ ] @yarvanya
- [x] @yurii-kokhalevych
- [ ] @denisbodnar

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

ToDo

## Summary of change

- created profile model (without associations and validations)
- added migration file for profile model
- added seed file for profile model

## Testing approach

ToDo
## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: server/seeders/20170523093919-unnamed-seeder.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        first_name: "Tom",
+        last_name: "Smith"
+      }
+    ], {});
+  },
+
+  down: function (queryInterface, Sequelize) {
+    return queryInterface.bulkDelete('Profiles', null, {});
+  }
+};
```
Comment (yarvanya) [inline server/seeders/20170523093919-unnamed-seeder.js]: Add new empty line

Comment (denisbodnar) [inline server/seeders/20170523093919-unnamed-seeder.js]: add a new line

File: server/models/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Object.keys(db).forEach(function(modelName) {
+  if (db[modelName].associate) {
+    db[modelName].associate(db);
+  }
+});
+
+db.sequelize = sequelize;
+db.Sequelize = Sequelize;
+
+module.exports = db;
```
Comment (yarvanya) [inline server/models/index.js]: This file has already created by Yuriy

File: server/migrations/20170522215056-create-profile.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+'use strict';
```
Comment (denisbodnar) [inline server/migrations/20170522215056-create-profile.js]: add an empty line

**Review by denisbodnar — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: server/seeders/profile.seed.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+    return queryInterface.bulkInsert('profiles', [
+      {
+        first_name: "John",
+        last_name: "Doe"
+      },
+      {
+        first_name: "Tom",
+        last_name: "Smith"
```
Comment (4asof4ik) [inline server/seeders/profile.seed.js]: Add info for user country....

File: server/routes/profile.router.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const profileController = require('../controllers').profiles;
+
+module.exports = (app) => {
+  app.get('/api', (req, res) => res.status(200).send({
+    message: 'Welcome.',
+  }));
+
+  app.post('/api/profiles', profileController.create);
+  app.get('/api/profiles', profileController.list);
```
Comment (4asof4ik) [inline server/routes/profile.router.js]: We have one profile for user

File: server/controllers/profile.controller.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    return profile
+    .create({
+      first_name: req.body.first_name,
+      last_name: req.body.last_name,
+    })
+    .then(profile => res.status(201).send(profile))
+    .catch(error => res.status(400).send(error));
+  },
+  
+  list(req, res) { // Display list of all profiles
```
Comment (4asof4ik) [inline server/controllers/profile.controller.js]: Drop this method

File: server/controllers/profile.controller.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        first_name: req.body.first_name || profiles.first_name,
+        last_name: req.body.last_name || profiles.last_name,
+      })
+      .then(() => res.status(200).send(profile)) // Send back the updated profile
+      .catch(error => res.status(400).send(error));
+    })
+    .catch(error => res.status(400).send(error));
+  },
+  
+  destroy(req, res){  // DELETE single profile
```
Comment (4asof4ik) [inline server/controllers/profile.controller.js]: Profile can be deleted only if user delet his account so this method can be dropped too

File: server/models/profile.model.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      type: DataTypes.STRING,
+      max: 200
+    }
+  }, {
+    timestamps: false,
+    createdAt: false,
+    updatedAt: false,
+    classMethods: {
+      associate: (models) => {
+        Profiles.belongsTo(models.users, {
```
Comment (4asof4ik) [inline server/models/profile.model.js]: Profile and User

File: server/models/profile.model.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+module.exports = function(sequelize, DataTypes) {
+  const profile = sequelize.define('profile', {
```
Comment (4asof4ik) [inline server/models/profile.model.js]: First latter should be in uppercase

File: server/models/profile.model.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    classMethods: {
+      associate: (models) => {
+        Profiles.belongsTo(models.users, {
+          foreignKey: 'profile_id',
+          onDelete: 'CASCADE',
+        });
+      },
+    },
+  });
+  return profile;
```
Comment (4asof4ik) [inline server/models/profile.model.js]: And here  uppercase

File: server/migrations/20170522215056-create-profile.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+        max: 200
+      },
+      country: {
+        type: DataTypes.STRING,
+        max: 200
+      }
+    });
+  },
+  down: function(queryInterface, Sequelize) {
+    return queryInterface.dropTable('profiles');
```
Comment (4asof4ik) [inline server/migrations/20170522215056-create-profile.js]: Profiles

File: server/migrations/20170522215056-create-profile.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    return queryInterface.createTable('profiles', {
```
Comment (4asof4ik) [inline server/migrations/20170522215056-create-profile.js]: Profiles

File: server/controllers/profile.controller.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const profile = require('../models').profile;
```
Comment (4asof4ik) [inline server/controllers/profile.controller.js]: Check if need use require('../models').Profile

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/profile.controller.js [hunk 14]
```diff
@@ -1,45 +1,34 @@
 'use strict';
 
-const Profile = require('../models/profile.model');
+// const Profile = require('../models/profile.model');
```
Comment (yurii-kokhalevych) [inline server/controllers/profile.controller.js]: drop comment

File: server/migrations/20170525184235-create-profile.js [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+        allowNull: false,
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.fn('now')
+      }
+    });
+  },
+  down: function(queryInterface, Sequelize) {
+    return queryInterface.dropTable('Profiles');
+  }
+};
```
Comment (yurii-kokhalevych) [inline server/migrations/20170525184235-create-profile.js]: add line

File: server/models/profile.model.js [hunk 16]
```diff
@@ -1,12 +1,12 @@
 'use strict';
 
 module.exports = function(sequelize, DataTypes) {
-  const Profile = sequelize.define('Profile', {
+  var Profile = sequelize.define('Profile', {
```
Comment (yurii-kokhalevych) [inline server/models/profile.model.js]: why var?

File: server/models/profile.model.js [hunk 17]
```diff
@@ -21,29 +21,24 @@ module.exports = function(sequelize, DataTypes) {
...
-          onDelete: 'CASCADE',
-        });
-      },
+      // associate: function(models) {
+      //   Profile.belongsTo(models.Users, {
+      //     foreignKey: 'id',
+      //     onDelete: 'CASCADE',
+      //   });
+      // },
     },
```
Comment (yurii-kokhalevych) [inline server/models/profile.model.js]: drop comments

File: server/routes/index.js [hunk 18]
```diff
@@ -0,0 +1,10 @@
+const profileController = require('../controllers/profile.controller');
+
+module.exports = (app) => {
+  app.get('/api', (req, res) => res.status(200).send({
+    message: 'Welcome to the DAMI API!',
+  }));
+
+  app.get('/api/profiles/:id', profileController.retrieve);
+  app.put('/api/profiles/:id', profileController.update);
+};
```
Comment (yurii-kokhalevych) [inline server/routes/index.js]: add line

File: server/models/profile.model.js (hunk 16 above)
Comment (yuliannakorol) [reply server/models/profile.model.js]: this was auto-generated by {{sequelize model:create}}

**Review by denisbodnar — DISMISSED** [review]

File: server/seeders/profile.seed.js [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+      {
+        first_name: "John",
+        last_name: "Doe",
+        birth_date: "",
+        address: "Street 1",
+        city: "Chicago",
+        country: "US",
+        createdAt: "2017-05-25 15:53:13.815+03",
+        updatedAt: "2017-05-25 15:53:13.815+03"
+
```
Comment (4asof4ik) [inline server/seeders/profile.seed.js]: drop

File: server/routes/index.js [hunk 20]
```diff
@@ -0,0 +1,10 @@
+const profileController = require('../controllers/profile.controller');
+
+module.exports = (app) => {
+  app.get('/api', (req, res) => res.status(200).send({
+    message: 'Welcome to the DAMI API!',
+  }));
+
+  app.get('/api/profiles/:id', profileController.retrieve);
```
Comment (4asof4ik) [inline server/routes/index.js]: should be profile because a user has one profile

File: server/app.js [hunk 21]
```diff
@@ -7,7 +7,7 @@ app.use(logger('dev'));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 
-require('./routes')(app);
+require('./routes/profile.router')(app);
```
Comment (4asof4ik) [inline server/app.js]: ??

File: server/models/profile.model.js [hunk 22]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+module.exports = function(sequelize, DataTypes) {
+  var Profile = sequelize.define('Profile', {
+    first_name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      max: 50
+      },
```
Comment (4asof4ik) [inline server/models/profile.model.js]: align

File: server/controllers/profile.controller.js [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+  
+  update(req, res) {
+    return Profile.findById(req.params.id)
+    .then(profile => {
+      if (!profile) {
+        return res.status(404).send({
+          message: 'Profile Not Found',
+        });
+      }
+      return profile.update({
```
Comment (4asof4ik) [inline server/controllers/profile.controller.js]: find the better way

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/profile.controller.js [hunk 24]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+// const Profile = require('../models/profile.model');
```
Comment (lhalam) [inline server/controllers/profile.controller.js]: delete comment

File: server/controllers/profile.controller.js [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+const Profile = require('../../config/db').Profile;
+
+module.exports = {
+  retrieve(req, res) {
+    return Profile.findById(req.params.id)
+    .then(profile => {
+      if (!profile) {
+        return res.status(404).send({ message: 'Profile Not Found'});
```
Comment (lhalam) [inline server/controllers/profile.controller.js:10]: remove extra spaces

File: server/controllers/profile.controller.js [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+      return res.status(200).send(profile);
+    })
+    .catch(error => res.status(400).send(error));
+  },
+  
+  update(req, res) {
+    return Profile.findById(req.params.id)
+    .then(profile => {
+      if (!profile) {
+        return res.status(404).send({
```
Comment (lhalam) [inline server/controllers/profile.controller.js]: comply with one style

File: server/models/profile.model.js [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+      type: DataTypes.STRING,
+      max: 100
+    },
+    country: {
+      type: DataTypes.STRING,
+      max: 100
+    },
+  }, {
+    classMethods: {
+      associate: function(models) {
```
Comment (lhalam) [inline server/models/profile.model.js]: empty implementation? is it necessary?

File: server/routes/profile.router.js [hunk 28]
```diff
@@ -0,0 +1,10 @@
+// const profileController = require('../controllers/profile.controller');
```
Comment (lhalam) [inline server/routes/profile.router.js]: delete comments

File: server/routes/index.js (hunk 20 above)
Comment (lhalam) [reply server/routes/index.js]: could be better use user_id?

File: server/controllers/profile.controller.js [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+  update(req, res) {
+    return Profile.findById(req.params.id)
+    .then(profile => {
+      
+      if (!profile) {
+        return res.status(404).send({message: 'Profile Not Found',
+        });
+      }
+      return profile.updateAttributes({
+        first_name: req.body.first_name || profile.first_name,
```
Comment (Vovk-Nazar) [inline server/controllers/profile.controller.js]: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

**Review by Vovk-Nazar — CHANGES_REQUESTED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Pull request closed without merging**